atomic_reactor/plugins/post_pulp_pull.py

```diff
@@ -60,9 +62,17 @@ def __init__(self, tasker, workflow,
         super(PulpPullPlugin, self).__init__(tasker, workflow)
         self.timeout = timeout
         self.retry_delay = retry_delay
-        self.insecure = insecure
-        self.secret = secret
-        self.expect_v2schema2 = expect_v2schema2
+
+        pulp_config = get_pulp(workflow, False)
+        if pulp_config:
+            pulp_handler = get_pulp_session(workflow, self.log, NO_FALLBACK)
+            self.secret = pulp_handler.get_secret()
+            self.insecure = pulp_handler.get_insecure()
+        else:
+            self.insecure = insecure
+            self.secret = secret
+
+        self.expect_v2schema2 = not get_prefer_schema1_digest(workflow, expect_v2schema2)
```

Comment:
Won't this negate `expect_v2schema2` as a fallback? Do we want this instead:
```python
self.expect_v2schema2 = not get_prefer_schema1_digest(workflow, not expect_v2schema2)
```

Reply:
Fixed.